src/store/reducers/appReducer.js

```diff
@@ -282,8 +284,13 @@ const updateSolutionSettingsReducer = (state, { payload, fromAction }) => update
 
 const updateActionSettingsReducer = (state, { payload, fromAction }) => {
   const { entities: { actionSettings } } = normalize(payload, actionSettingsListSchema);
+  var applicationPermissionsAssigned = undefined;
```

Comment:
Consider using 
```js
dot.pick('Email.applicationPermissionsAssigned', actionSettings);
```
to avoid the null check. Reduces lines of code. Similar usage in `pollActionSettings` in this same file.